lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -40,86 +39,49 @@ sub _git_clone_all ($job, $clones) {
 
     # iterate clones sorted by path length to ensure that a needle dir is always cloned after the corresponding casedir
     for my $path (sort { length($a) <=> length($b) } keys %$clones) {
+        my $git = OpenQA::Git->new(app => $app, dir => $path);
         my $url = $clones->{$path};
         die "Don't even think about putting '..' into '$path'." if $path =~ /\.\./;
-        eval { _git_clone($job, $ctx, $path, $url) };
+        eval { _git_clone($git, $job, $ctx, $path, $url) };
```

Comment:
I think we can create the `$git` object in `_git_clone` because we only need it there and not in the loop here.
One less parameter to pass around.